### PR TITLE
High-Order-Components. Closes #24

### LINKED ISSUE #24: Higher Order Components  (HOC)

### PR DESCRIPTION
**Pull request by fosteman**

### CODE REVIEW

File: content/blog/hoc/index.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+title: High-Order Components
+date: "2019-06-03"
+author: fosteman
+---
+
+# High-Order Components
+
+An old saying "Don't Repeat Yourself" or DRY in React knowledge base and codebase is a worthwhile challenge.
+High-Order Components are functions wrapping React Components solving this incongruence.
+For example [.map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)
```
Comment (KirankumarAmbati) [inline content/blog/hoc/index.md]: I didn't get this part @fosteman. (reg. map function)

File: content/blog/hoc/index.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+```javascript
+const reverse = childComponent =>
+  ({ children, ...props }) =>
+    <childComponent {...props}>
+      {children.split("").reverse().join("")}
+    </childComponent>
+const Palindrom = props => <span>{props.children}</span>
+const reversedPalindrom = reverse(name)
+
+assert_equals(Palindrom, reversedPalindrom)
```
Comment (KirankumarAmbati) [inline content/blog/hoc/index.md]: Here, are we assuming that **assert_equals** is a function that checks equality of two DOM elements ?

Comment (fosteman) [reply content/blog/hoc/index.md]: I guess I was little drunken when wrote this. Gonna rework this example

File: content/blog/hoc/index.md (hunk 1 above)
Comment (fosteman) [reply content/blog/hoc/index.md]: Would `Array.map`(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) be better?

Comment (KirankumarAmbati) [reply content/blog/hoc/index.md]: My question is like is .map is an example of Higher Order Function ?

Comment (fosteman) [reply content/blog/hoc/index.md]: There is a built-in array method, forEach, that provides something like a for/of loop as a higher-order function.
```
["A", "B"].forEach(l => console.log(l));
```
Will be clarified, thanks for reviewing, @KirankumarAmbati !

Comment (KirankumarAmbati) [reply content/blog/hoc/index.md]: Thanks for the clarification! @fosteman :)

File: .gitignore [hunk 3]
```diff
@@ -67,3 +67,10 @@ yarn-error.log
 .pnp.js
 # Yarn Integrity file
 .yarn-integrity
+.idea/workspace.xml
```
Comment (KirankumarAmbati) [inline .gitignore]: Just curious about these additions. Please can you help me understand their usage.

**Comment by KirankumarAmbati** [comment]
@fosteman, Thanks for the PR! :)

**Pull request merged**